### PR TITLE
Add ESPHome native API docs

### PR DESCRIPTION
**Pull request by OttoWinter**
**Description:**

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#19334

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_components/esphome.markdown [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+This component allows you to connect your [ESPHome](https://esphomelib.com/esphomeyaml/index.html) devices directly into Home Assistant with the [native ESPHome API](https://esphomelib.com/esphomeyaml/components/api.html).
+
+## {% linkable_title Setup the component via the integrations screen %}
+
+Menu: *Configuration* -> *Integrations*
+
+Press on **ESPHome** and configure the integration:
+
+* Enter your the **address** and **port** of the ESP. For example if the node is called `livingroom`, the address would be `livingroom.local` and the port number `6053` (default).
```
Comment (MartinHjelmare) [inline source/_components/esphome.markdown]: Enter the address and port of your ESP.

File: source/_components/esphome.markdown [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+description: "Support for ESPHome devices using the native ESPHome API."
+date: 2018-12-16 14:20
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: esphome.png
+ha_category: DIY
+ha_release: 0.85
+---
```
Comment (MartinHjelmare) [inline source/_components/esphome.markdown:14]: Add a `ha_iot_class`.

File: source/_components/esphome.markdown [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+This component allows you to connect your [ESPHome](https://esphomelib.com/esphomeyaml/index.html) devices directly into Home Assistant with the [native ESPHome API](https://esphomelib.com/esphomeyaml/components/api.html).
+
+## {% linkable_title Setup the component via the integrations screen %}
+
+Menu: *Configuration* -> *Integrations*
+
+Press on **ESPHome** and configure the integration:
+
+* Enter your the address and port of your ESP. For example if the node is called `livingroom`, the address would be `livingroom.local` and the port number `6053` (default).
```
Comment (MartinHjelmare) [inline source/_components/esphome.markdown]: Grammar of first sentence is still not correct. Change to something like:

"Enter the address and port of your ESP."

Comment (OttoWinter) [reply source/_components/esphome.markdown]: Oh now I know what you mean 🤕  Thanks!

**Review by MartinHjelmare — APPROVED** [review]
Nice! Can be merged when build passes.

**Review by fabaff — APPROVED** [review]

**Pull request merged**